pallets/asset-registry/src/migration.rs

```diff
@@ -7,81 +7,170 @@
 	log,
 	traits::{Get, StorageVersion},
 	weights::Weight,
+	Twox64Concat,
 };
 
-///
+use crate::*;
+
 pub mod v1 {
 	use super::*;
 	use codec::{Decode, Encode};
+	use frame_support::storage_alias;
 	use scale_info::TypeInfo;
 	use sp_core::RuntimeDebug;
+	use sp_runtime::BoundedVec;
 
 	#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, TypeInfo)]
-	pub struct OldAssetDetails<AssetId, Balance, BoundedString> {
-		/// The name of this asset. Limited in length by `StringLimit`.
-		pub(super) name: BoundedString,
+	pub struct AssetDetails<AssetId, Balance, BoundedString> {
+		pub name: BoundedString,
+		pub asset_type: AssetType<AssetId>,
+		pub existential_deposit: Balance,
+		pub xcm_rate_limit: Option<Balance>,
+	}
 
-		pub(super) asset_type: AssetType<AssetId>,
+	#[derive(Clone, Encode, Decode, Eq, PartialEq, Default, RuntimeDebug, TypeInfo)]
+	pub struct AssetMetadata<BoundedString> {
+		pub(super) symbol: BoundedString,
+		pub(super) decimals: u8,
+	}
 
-		pub(super) existential_deposit: Balance,
+	#[storage_alias]
+	pub type Assets<T: Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		<T as crate::Config>::AssetId,
+		AssetDetails<<T as crate::Config>::AssetId, Balance, BoundedVec<u8, <T as crate::Config>::StringLimit>>,
+		OptionQuery,
+	>;
+
+	#[storage_alias]
+	pub type AssetMetadataMap<T: Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		<T as crate::Config>::AssetId,
+		AssetMetadata<BoundedVec<u8, <T as crate::Config>::StringLimit>>,
+		OptionQuery,
+	>;
+
+	#[storage_alias]
+	pub type AssetLocations<T: Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		<T as crate::Config>::AssetId,
+		<T as crate::Config>::AssetNativeLocation,
+		OptionQuery,
+	>;
+}
 
-		pub(super) locked: bool,
-	}
+pub mod v2 {
+	use super::*;
 
 	pub fn pre_migrate<T: Config>() {
-		assert_eq!(StorageVersion::get::<Pallet<T>>(), 0, "Storage version too high.");
+		assert_eq!(StorageVersion::get::<Pallet<T>>(), 1, "Storage version too high.");
 
 		log::info!(
 			target: "runtime::asset-registry",
 			"Asset Registry migration: PRE checks successful!"
 		);
 	}
 
-	pub fn migrate<T: Config>() -> Weight {
+	pub fn migrate<T: Config<AssetId = u32>>() -> Weight {
+		log::info!(
+			target: "runtime::asset-registry",
+			"Running migration to v2 for Asset Registry"
+		);
+
 		log::info!(
 			target: "runtime::asset-registry",
-			"Running migration to v1 for Asset Registry"
+			"Migrating Assets storage"
 		);
 
 		let mut i = 0;
-		Assets::<T>::translate(
-			|_key,
-			 OldAssetDetails {
-			     name,
-			     asset_type,
-			     existential_deposit,
-			     locked: _,
-			 }| {
-				i += 1;
-				Some(AssetDetails {
-					name,
-					asset_type,
-					existential_deposit,
-					xcm_rate_limit: None,
-				})
-			},
+		let mut v2_assets_details = Vec::<(
+			<T as crate::Config>::AssetId,
+			AssetDetails<<T as crate::Config>::AssetId, <T as crate::Config>::StringLimit>,
+		)>::new();
+		for (k, v) in v1::Assets::<T>::iter() {
```

Comment:
you don't remove `v1::Assets`?

Reply:
Based on the doc iter+remove is [undefined result](https://docs.rs/frame-support/latest/frame_support/storage/trait.IterableStorageDoubleMap.html#tymethod.iter) so I didn't want not risk it. 
All the elements are overwritten here:
https://github.com/galacticcouncil/HydraDX-node/blob/f0e4412cc6358c60247ba40f08e83a8b96f13aa6/pallets/asset-registry/src/migration.rs#L129C3-L136C4

Reply:
yeah, that's where you should remove.
At the point where you insert into v2 the iteration of v1 is done